### PR TITLE
[catalog/unprocessed-entities] add sort and filter

### LINKED ISSUE #18694: 🐛 Bug Report: catalog-unprocessed-entities plugin: Filter and column sorting doesn't work
### 📜 Description

After installing the catalog-unprocessed-entities plugin and the catalog-backend-module-unprocessed plugin and then navigating to the catalog-unprocessed-entities page, the filter and column sorting doesn't work.
When I try to enter any letters into the filter, I get no results in the unprocessed entities list even though there are items that match those letters.
Column sorting also doesn't work. If I select the arrow next to ENTITYREF, KIND, OWNER, or RAW labels, the order of items doesn't change.

### 👍 Expected behavior

When I type letters into the Filter search bar, I should see results show up that match those letters.
When I click the arrow next to the head of a column, that should sort the column's items.

### 👎 Actual Behavior with Screenshots

When I type letters into the Filter search bar, no items show up. The results are completely empty even though there are items that match the letters.
When I click the arrow next to the head of a column, the list of items stays the same and doesn't get sorted.

### 👟 Reproduction steps

1. Install and configure the [catalog-backend-module-unprocessed plugin](https://github.com/backstage/backstage/tree/v1.15.0/plugins/catalog-backend-module-unprocessed) 
2. Install and configure the [catalog-unprocessed-entities plugin](https://github.com/backstage/backstage/tree/v1.15.0/plugins/catalog-unprocessed-entities)
3. Start your application locally
4. Navigate to [/catalog-unprocessed-entities](http://localhost:3000/catalog-unprocessed-entities)
5. You will need a list of unprocessed entities already showing up under the Failed tab in order to test the filter
6. Type any letter into the Filter search bar to see the filter issue
7. Click the arrow next to the ENTITYREF, KIND, OWNER, or RAW labels to see the column sorting issue

### 📃 Provide the context for the Bug.

_No response_

### 🖥️ Your Environment

node: v16.14.2
yarn: 1.22.19
backstage:  1.15.0

Dependencies:
  @backstage/plugin-catalog-backend-module-unprocessed  0.1.0
  @backstage/plugin-catalog-unprocessed-entities: 0.1.0

### 👀 Have you spent some time to check if this bug has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/backstage/backstage/blob/master/CODE_OF_CONDUCT.md)

### Are you willing to submit PR?

None

### PR DESCRIPTION
**Pull request by alde**
Implement sorting and filtering to unprocessed entities tables.

Sorting added to entity-ref, owner and kind columns.

Filtering/searching added to entity-ref column.

Fixes #18694

#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [ ] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: packages/catalog-model/examples/apis/petstore-webhook-api.yaml [hunk 1]
```diff
@@ -16,6 +16,6 @@ metadata:
 spec:
   type: openapi
   lifecycle: experimental
-  owner: team-c
```
Comment (benjdlambert) [inline packages/catalog-model/examples/apis/petstore-webhook-api.yaml:19]: Do we need this change? Can maybe remove this and the changeset for the package?

Comment (alde) [reply packages/catalog-model/examples/apis/petstore-webhook-api.yaml:19]: I guess not, it was to test so the sorting worked for a nested key

File: plugins/catalog-unprocessed-entities/src/components/PendingEntities.tsx [hunk 2]
```diff
@@ -84,6 +96,9 @@ export const PendingEntities = () => {
         options={{ pageSize: 20 }}
         columns={columns}
         data={data?.entities || []}
+        onSearchChange={(searchTerm: string) =>
+          setSelectedSearchTerm(searchTerm)
```
Comment (freben) [inline plugins/catalog-unprocessed-entities/src/components/PendingEntities.tsx:100]: is this here only to trigger a repaint of this component? is it really needed? we don't read the state after all

File: plugins/catalog-unprocessed-entities/src/components/PendingEntities.tsx [hunk 3]
```diff
@@ -57,22 +58,33 @@ export const PendingEntities = () => {
   const columns: TableColumn[] = [
     {
       title: <Typography>entityRef</Typography>,
+      sorting: true,
+      field: 'entity_ref',
+      customFilterAndSearch: (query, row: any) =>
+        row.entity_ref
+          .toLocaleUpperCase('en-US')
+          .includes(query.toLocaleUpperCase('en-US')),
```
Comment (freben) [inline plugins/catalog-unprocessed-entities/src/components/PendingEntities.tsx:66]: does it make sense to make the other fields case insensitive too?

i actually thought it was case insensitive by default 🤔 

(comment applies to the other file as well)

**Review by freben — APPROVED** [review]

File: plugins/catalog-unprocessed-entities/src/components/PendingEntities.tsx (hunk 2 above)
Comment (alde) [reply plugins/catalog-unprocessed-entities/src/components/PendingEntities.tsx:100]: yes, copied from how other Table components deal with search/filter

**Pull request merged**